lib/board.rb

```diff
@@ -7,7 +7,23 @@ def initialize(spaces = (1..9).to_a)
     @spaces = spaces
   end
 
-  def record_move(marker, index)
-    spaces[index] = marker
+  def record_move(marker, selection)
+    index = convert(selection)
+    @spaces[index] = marker
+  end
+
+  def full?(player1_marker, player2_marker)
+    @spaces.all? { |value| value == player1_marker || value == player2_marker }
+  end
+
+  def occupied?(selection, player1_marker, player2_marker)
+    index = convert(selection)
+    @spaces[index] == player1_marker || @spaces[index] == player2_marker
+  end
```

Comment:
Do you have tests for `full?` and `occupied?` methods? And is there any ways to check if a board is full or a place is occupied without providing the players markers to the Board?